Link errors on OS X with PostgreSQL 9.4

When trying to install on OS X with PostgreSQL 9.4 (tried both http://postgresapp.com and http://www.enterprisedb.com/products-services-training/pgdownload), I get a wall of link errors. I'm assuming I'm missing a library, but I can't figure out what library that is.
```
gcc -mmacosx-version-min=10.7 -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv  -bundle -multiply_defined suppress -o multicorn.so src/errors.o src/python.o src/query.o src/multicorn.o -L/Applications/Postgres.app/Contents/Versions/9.4/lib -L/Applications/Postgres.app/Contents/Versions/9.4/lib  -L/Applications/Postgres.app/Contents/Versions/9.4/lib -Wl,-dead_strip_dylibs   -L/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config -ldl -framework CoreFoundation -lpython2.7 -ldl  -framework CoreFoundation
Undefined symbols for architecture x86_64:
  "_AllocSetContextCreate", referenced from:
      _getCacheEntry in python.o
  "_CacheMemoryContext", referenced from:
      _getCacheEntry in python.o
      __PG_init in multicorn.o
  "_CurrentMemoryContext", referenced from:
      _getCacheEntry in python.o
      _MemoryContextSwitchTo in python.o
      _canonicalScalarArrayOpExpr in query.o
      _findPaths in query.o
      _MemoryContextSwitchTo in multicorn.o
      _multicorn_handler in multicorn.o
  "_DirectFunctionCall1Coll", referenced from:
      _datumNumberToPython in python.o
      _datumDateToPython in python.o
....
```

--- a/Makefile
+++ b/Makefile
@@ -90,6 +90,9 @@ else
 	override CPPFLAGS := $(PG_CPPFLAGS) $(CPPFLAGS)
 endif
 
+ifeq ($(PORTNAME),darwin)
+	override LDFLAGS += -undefined dynamic_lookup -bundle_loader $(shell $(PG_CONFIG) --bindir)/postgres
+endif
 
 PYTHON_TEST_VERSION ?= $(python_version)
 PG_TEST_VERSION ?= $(MAJORVERSION)

